MaxText/layers/linears.py

```diff
@@ -247,12 +247,12 @@ def __call__(self, inputs, decode: bool = False, deterministic: bool = False):
         )(inputs)
         if cfg.activations_in_float32:
           x = x.astype(jnp.float32)
+        x = checkpoint_name(x, "mlp" + dense_name)
```

Comment:
I think we want this before the float32 cast to save 2x memory in the float32 case

Reply:
For some reason I can't comment on lines above - but we also should add a checkpoint_name after line 230 in the fused case (can be mlpwi or mlpwi_fused)

Reply:
Its somewhat ugly to have mlpwi_0 and mlpwi_1, we could re-use the same name which makes for prettier code but I can see some cases where we might actually want to do different things with the two of them since they are  so huge...

Reply:
moved and assigned a checkpoint name for fused mlp